### PR TITLE
Upgrade to Scala 2.12.6

### LINKED ISSUE #1515: release for scala 2.12.6

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #1515, see commit history for details.

### CODE REVIEW

**Comment by fommil** [comment]
thank you!

File: tests/jvm/src/test/scala/scala/meta/tests/trees/ReflectionSuite.scala [hunk 1]
```diff
@@ -50,33 +51,36 @@ class ReflectionSuite extends FunSuite {
     // NOTE: we can't just do `duplicateRelevantFieldTpes.distinct`, because that doesn't account for `=:=`
     val distinctRelevantFieldTpes = ListBuffer[Type]()
     duplicateRelevantFieldTpes.foreach(tpe => if (!distinctRelevantFieldTpes.exists(_ =:= tpe)) distinctRelevantFieldTpes += tpe)
-    assert(distinctRelevantFieldTpes.sortBy(_.toString).mkString(EOL) === """
+    val obtained = distinctRelevantFieldTpes.sortBy(_.toString).mkString(EOL)
```
Comment (olafurpg) [inline tests/jvm/src/test/scala/scala/meta/tests/trees/ReflectionSuite.scala]: This test now fails on 2.11, I propose we move this test to 2.12 only or we remove the test suite altoghther. wdyt @xeno-by ?

Comment (xeno-by) [reply tests/jvm/src/test/scala/scala/meta/tests/trees/ReflectionSuite.scala]: Let's move to 2.12 only.

Comment (olafurpg) [reply tests/jvm/src/test/scala/scala/meta/tests/trees/ReflectionSuite.scala]: Done.

**Pull request merged**